idseq_dag/util/s3.py

```diff
@@ -96,7 +97,7 @@ def touch_s3_file_list(s3_dir, file_list):
         touch_s3_file(os.path.join(s3_dir, f))
 
 
-def install_s3mi(installed={}, mutex=TraceLock("install_s3mi", threading.RLock())):  # pylint: disable=dangerous-default-value
+def install_s3mi(installed={}, mutex=TraceLock("install_s3mi", multiprocessing.RLock())):  # pylint: disable=dangerous-default-value
```

Comment:
Multiprocessing locks work equally well when using threads and processes.    We don't really plan to call this function from multiple processes (we recommend multiple threads), but inevitably, someone might do that, or might even have done that unintentionally.  To harden against that possibility, we should switch all threading lock to multiprocessing locks.